### PR TITLE
use instanceIDForTest for instance init in unit tests

### PR DESCRIPTION
**Pull request by chliangGoogle**
Original init method for unit test could return nil object with the new component model.

### CODE REVIEW

File: Example/Messaging/Tests/FIRMessagingTest.m [hunk 1]
```diff
@@ -55,6 +61,7 @@ @implementation FIRMessagingTest
 - (void)setUp {
   [super setUp];
   _messaging = [FIRMessaging messagingForTests];
+  _messaging.instanceID = [FIRInstanceID instanceIDForTests];
```
[the remark this answers is not in the document]
Comment (chliangGoogle) [reply Example/Messaging/Tests/FIRMessagingTest.m]: Yes, I think each test use those mock for stubbing method.

The travis doesn't run right now cuz it doesn't have the latest iid.

[the remark this answers is not in the document]
Comment (chliangGoogle) [reply Example/Messaging/Tests/FIRMessagingTest.m]: actually we can get rid of realInstanceID

**Comment by chliangGoogle** [comment]
Because it doesn't have the internal change yet in IID, what should we do in this scenario?

**Comment by paulb777** [comment]
Oh right, I missed that.

We should put the new InstanceID podspec on the staging spec repo.  I'll set that up.

**Comment by paulb777** [comment]
The tests are now fixed with SpecsStaging to get the future InstanceID pod.

**Pull request merged**